src/main/scala/elevate/rise/rules/algorithmic.scala

```diff
@@ -107,4 +108,278 @@ object algorithmic {
       Success(oclSlideSeq(rot)(a)(sz)(sp)(wr)(typed(f) >> g)(e) :: expr.t)
     case _ => Failure(slideSeqFusion)
   }
+
+  // overlapped tiling
+
+  // constraint: n - m = u - v
+  // v = u + m - n
+  def slideOverlap(u: Nat): Strategy[Rise] =
+    `slide(n, m) -> slide(u, v) >> *(slide(n, m)) >> J`(u)
+  def `slide(n, m) -> slide(u, v) >> *(slide(n, m)) >> J`(
+   u: Nat
+  ): Strategy[Rise] = {
+    case expr @ DepApp(DepApp(Slide(), n: Nat), m: Nat) =>
+      val v = u + m - n
+      Success((slide(u)(v) >> map(slide(n)(m)) >> join) :: expr.t)
+    case _ =>
+      Failure(slideOverlap(u))
+  }
+
+  // slide widening
+
+  def dropInSlide: Strategy[Rise] =
+    `slide n 1 >> drop l -> slide (n+l) 1 >> map(drop l)`
+  def `slide n 1 >> drop l -> slide (n+l) 1 >> map(drop l)`: Strategy[Rise] = {
+    case expr @ App(DepApp(Drop(), l: Nat),
+      App(DepApp(DepApp(Slide(), n: Nat), Cst(1)), in)
+    ) =>
+      Success(app(map(drop(l)), app(slide(n+l)(1), typed(in))) :: expr.t)
+    case _ =>
+      Failure(dropInSlide)
+  }
+
+  def takeInSlide: Strategy[Rise] =
+    `slide n 1 >> take (N - r) -> slide (n+r) 1 >> map(take (n - r))`
+  def `slide n 1 >> take (N - r) -> slide (n+r) 1 >> map(take (n - r))`: Strategy[Rise] = {
+    case expr @ App(t @ DepApp(Take(), rem: Nat),
+    App(DepApp(DepApp(Slide(), n: Nat), Cst(1)), in)
+    ) =>
+      t.t match {
+        case FunType(ArrayType(size, _), _) =>
+          val r = size - rem
+          Success(app(map(take(n)), app(slide(n+r)(1), typed(in))) :: expr.t)
+        case _ => throw new Exception("this should not happen")
+      }
+    case _ =>
+      Failure(dropInSlide)
+  }
+
+  def dropNothing: Strategy[Rise] = {
+    case expr @ DepApp(Drop(), Cst(0)) => Success(fun(x => x) :: expr.t)
+    case _ => Failure(dropNothing)
+  }
+
+  def takeAll: Strategy[Rise] = {
+    case expr @ DepApp(Take(), n: Nat) => expr.t match {
+      case FunType(ArrayType(m, _), _) if n == m => Success(fun(x => x) :: expr.t)
+      case _ => Failure(takeAll)
+    }
+    case _ => Failure(takeAll)
+  }
+
+  def mapIdentity: Strategy[Rise] = {
+    case expr @ App(Map(), Lambda(x1, x2)) if x1 == x2 => Success(fun(x => x) :: expr.t)
+    case _ => Failure(mapIdentity)
+  }
+
+  def slideAfter: Strategy[Rise] = `x -> join (slide 1 1 x)`
+  def `x -> join (slide 1 1 x)`: Strategy[Rise] = {
+    x => Success(join(slide(1)(1)(x)) :: x.t)
+  }
+
+  def slideAfter2: Strategy[Rise] = {
```

Comment:
I have this second variant because I needed a `map` pattern after applying this rule .. but maybe this can be a strategy changing the `join` ..